ReActAgentState.fromJSON() 反序列化失败：fastjson2 无法实例化抽象类 Message

### Bug Description

## 环境
- agents-flex 版本：2.1.3
- fastjson2 版本：2.0.58（agents-flex 依赖） / 2.0.60（项目依赖）
- JDK 21
## 复现步骤
```java
ReActAgent agent = new ReActAgent(chatModel, tools, "北京天气如何？");
agent.execute();
// toJSON 正常
String json = agent.getState().toJSON();
System.out.println(json);  // JSON 中包含 @type 注解
// fromJSON 抛异常
ReActAgentState restored = ReActAgentState.fromJSON(json);
异常堆栈
com.alibaba.fastjson2.JSONException: create instance error, class com.agentsflex.core.message.Message
    at com.alibaba.fastjson2.reader.ObjectReaderAdapter.createInstance(ObjectReaderAdapter.java:441)
    at com.alibaba.fastjson2.reader.ObjectReaderBean.readObject(ObjectReaderBean.java:386)
    at com.alibaba.fastjson2.JSON.parseObject(JSON.java:1171)
    at com.agentsflex.core.agent.react.ReActAgentState.fromJSON(ReActAgentState.java:108)
Caused by: java.lang.InstantiationException
    ... (Message 是抽象类，无法直接实例化)
原因分析
toJSON() 使用了 JSONWriter.Feature.WriteClassName，JSON 输出中确实包含 @type 信息：

{"@type":"com.agentsflex.core.message.UserMessage","content":"..."}
但 fromJSON() 使用的 JSONReader.Feature.SupportClassForName 在处理 List<Message> 中的元素时，没有正确利用 @type 信息来实例化具体子类（UserMessage、AiMessage），而是尝试直接实例化抽象类 Message，导致 InstantiationException。

临时绕过方案
目前只能手动序列化消息历史，绕过 fromJSON：

// 保存：手动提取消息历史
List<Map<String,String>> history = new ArrayList<>();
for (Message m : agent.getMemoryPrompt().getMessages()) {
    String role = (m instanceof AiMessage) ? "assistant" : "user";
    String content = (m instanceof AiMessage ai) ? ai.getFullContent() : ...;
    history.add(Map.of("role", role, "content", content));
}
String json = JSONUtil.toJson(history);

// 恢复：手动重建 Message 列表
List<Message> messages = parseHistory(json);
ReActAgentState state = new ReActAgentState();
state.setMessageHistory(messages);
state.addMessage(new UserMessage(userReply));
ReActAgent agent = new ReActAgent(chatModel, tools, state);
影响
状态恢复是 ReAct Agent 交互式追问的核心机制。此问题导致 fromJSON() 完全不可用，无法实现"agent 反问 → 用户回复 → 从中断处继续推理"的闭环。

### Dependency Updates

- [x] Yes, I've updated all my dependencies.

--- a/agents-flex-core/src/main/java/com/agentsflex/core/agent/react/ReActAgentState.java
+++ b/agents-flex-core/src/main/java/com/agentsflex/core/agent/react/ReActAgentState.java
@@ -105,7 +105,11 @@ public class ReActAgentState implements Serializable {
     }
 
     public static ReActAgentState fromJSON(String json) {
-        return JSON.parseObject(json, ReActAgentState.class, JSONReader.Feature.SupportClassForName);
+        // toJSON() writes @type markers via JSONWriter.Feature.WriteClassName. Reading them back to
+        // restore the concrete Message subtypes (UserMessage / AiMessage ...) in messageHistory
+        // requires SupportAutoType — its counterpart. SupportClassForName only covers Class-typed
+        // fields and leaves fastjson2 trying to instantiate the abstract Message, which fails.
+        return JSON.parseObject(json, ReActAgentState.class, JSONReader.Feature.SupportAutoType);
     }
 
 

--- a/agents-flex-core/src/test/java/com/agentsflex/core/agent/react/ReActAgentStateTest.java
+++ b/agents-flex-core/src/test/java/com/agentsflex/core/agent/react/ReActAgentStateTest.java
@@ -1,0 +1,102 @@
+/*
+ *  Copyright (c) 2023-2026, Agents-Flex (fuhai999@gmail.com).
+ *  <p>
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *  <p>
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *  <p>
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package com.agentsflex.core.agent.react;
+
+import com.agentsflex.core.message.AiMessage;
+import com.agentsflex.core.message.Message;
+import com.agentsflex.core.message.UserMessage;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.Assert.*;
+
+/**
+ * 复现并验证 GitHub issue #50：ReActAgentState 的 toJSON()/fromJSON() 往返序列化。
+ *
+ * <p>修复前，{@link ReActAgentState#fromJSON(String)} 使用 {@code SupportClassForName}，
+ * 无法解析 {@code @type} 标记，fastjson2 会尝试实例化抽象类 {@link Message} 并抛出
+ * {@code create instance error}。修复后使用 {@code SupportAutoType}，可正确还原
+ * {@link UserMessage}/{@link AiMessage} 等具体子类型。
+ *
+ * <p>本测试不依赖网络或 LLM，可离线运行。
+ */
+public class ReActAgentStateTest {
+
+    @Test
+    public void testToJsonFromJsonRoundTrip_restoresConcreteMessageSubtypes() {
+        ReActAgentState state = new ReActAgentState();
+        state.setUserQuery("今天的天气怎么样？");
+        state.setIterationCount(3);
+        state.setMaxIterations(20);
+        state.setStreamable(true);
+        state.setContinueOnActionInvokeError(true);
+
+        UserMessage userMessage = new UserMessage("我在北京市");
+        userMessage.putMetadata("type", "reActObservation");
+
+        AiMessage aiMessage = new AiMessage();
+        aiMessage.setContent("Final Answer: 北京今天晴。");
+        aiMessage.putMetadata("type", "reActFinalAnswer");
+
+        List<Message> history = new ArrayList<>();
+        history.add(userMessage);
+        history.add(aiMessage);
+        state.setMessageHistory(history);
+
+        // 序列化 -> 反序列化（修复前此处会抛出 JSONException: create instance error）
+        String json = state.toJSON();
+        ReActAgentState restored = ReActAgentState.fromJSON(json);
+
+        assertNotNull(restored);
+
+        // 标量字段往返保持
+        assertEquals("今天的天气怎么样？", restored.getUserQuery());
+        assertEquals(3, restored.getIterationCount());
+        assertEquals(20, restored.getMaxIterations());
+        assertTrue(restored.isStreamable());
+        assertTrue(restored.isContinueOnActionInvokeError());
+
+        // 关键：messageHistory 中的具体子类型被正确还原
+        List<Message> restoredHistory = restored.getMessageHistory();
+        assertNotNull(restoredHistory);
+        assertEquals(2, restoredHistory.size());
+
+        assertTrue("第一个消息应还原为 UserMessage", restoredHistory.get(0) instanceof UserMessage);
+        assertTrue("第二个消息应还原为 AiMessage", restoredHistory.get(1) instanceof AiMessage);
+
+        UserMessage restoredUser = (UserMessage) restoredHistory.get(0);
+        assertEquals("我在北京市", restoredUser.getContent());
+        assertEquals("reActObservation", restoredUser.getMetadata("type"));
+
+        AiMessage restoredAi = (AiMessage) restoredHistory.get(1);
+        assertEquals("Final Answer: 北京今天晴。", restoredAi.getContent());
+        assertEquals("reActFinalAnswer", restoredAi.getMetadata("type"));
+    }
+
+    @Test
+    public void testFromJson_emptyHistory_roundTrips() {
+        ReActAgentState state = new ReActAgentState();
+        state.setUserQuery("hello");
+
+        ReActAgentState restored = ReActAgentState.fromJSON(state.toJSON());
+
+        assertNotNull(restored);
+        assertEquals("hello", restored.getUserQuery());
+        assertNull(restored.getMessageHistory());
+    }
+}

